### ISSUE TITLE
Import does not actually import the data

### ISSUE THREAD

**Issue by peterbaumert** [CiscoDevNet/terraform-provider-iosxe#134]
Hi,

since I am still new to terraform, I am not 100% certain if this is actually how its intended to work. So just close the issue if all works as designed.

When I do a 

```
terraform import iosxe_system.hostname "Cisco-IOS-XE-native:native"
iosxe_system.hostname: Importing from ID "Cisco-IOS-XE-native:native"...
iosxe_system.hostname: Import prepared!
  Prepared iosxe_system for import
iosxe_system.hostname: Refreshing state... [id=Cisco-IOS-XE-native:native]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

All seems to have worked fine, but when I now want to see the state it just shows the `id`:

```
terraform state show iosxe_system.hostname
# iosxe_system.hostname:
resource "iosxe_system" "hostname" {
    id = "Cisco-IOS-XE-native:native"
}
```

From other provides I know that usually there should be the current data which is configured in the imported device.

You can also see this as a feature request, if this is just not implemented yet :)

KR
Peter

**Comment by danischm** [comment]
How did the resource config look like or did you not have a corresponding resource in your config and were using the `-generate-config-out` cli flag?

**Comment by peterbaumert** [comment]
resource looked like

```
resource "iosxe_system" "hostname" {
  hostname = "hostname"
}
```

the -generate-config-out flag does not seem to do anything either.

```
Terraform v1.7.4
on linux_amd64
+ provider registry.terraform.io/ciscodevnet/iosxe v0.5.5
```

I mean it does import the resource in general, I just does not have the actual values from the device in them and wants to update them although nothing changed.

**Comment by NWA2000** [comment]
Pretty sure this isn't refreshing the state of the device either (IE not reading actuals the device).  Making manual configuration drift (IE changing an interface description on the switch) results in no terraform changes of an existing resource.

**Comment by peterbaumert** [comment]
Aby thoughts when this issue will be looked at? Kind of a showstopper for me :/

**Comment by danischm** [comment]
Import should be fully supported from v0.5.8 (just published).

**Comment by peterbaumert** [comment]
> Import should be fully supported from v0.5.8 (just published).

Will try first thing tomorrow and report back. Thanks

**Comment by NWA2000** [comment]
Seems to work for me:

`terraform version
Terraform v1.11.2
on darwin_arm64
+ provider registry.terraform.io/ciscodevnet/iosxe v0.5.8`

Have to use the ascii character for the slash however (%2F as the slash in the ID)

`terraform import 'iosxe_interface_ethernet.interfaces["1/0/42"]' "Cisco-IOS-XE-native:native/interface/GigabitEthernet=1%2F0%2F42"`


Plan does change false values to null, but I can adapt my definition.

`terraform plan
iosxe_interface_ethernet.interfaces["1/0/42"]: Refreshing state... [id=Cisco-IOS-XE-native:native/interface/GigabitEthernet=1%2F0%2F42]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # iosxe_interface_ethernet.interfaces["1/0/42"] will be updated in-place
  ~ resource "iosxe_interface_ethernet" "interfaces" {
      - authentication_order_dot1x                 = false -> null
      - authentication_order_dot1x_mab             = false -> null
      - authentication_order_dot1x_webauth         = false -> null
      - authentication_order_mab                   = false -> null
      - authentication_order_mab_dot1x             = false -> null
      - authentication_order_mab_webauth           = false -> null
      - authentication_order_webauth               = false -> null
      - authentication_periodic                    = false -> null
      - authentication_priority_dot1x              = false -> null
      - authentication_priority_dot1x_mab          = false -> null
      - authentication_priority_dot1x_webauth      = false -> null
      - authentication_priority_mab                = false -> null
      - authentication_priority_mab_dot1x          = false -> null
      - authentication_priority_mab_webauth        = false -> null
      - authentication_priority_webauth            = false -> null
      - authentication_timer_reauthenticate_server = false -> null
      - auto_qos_classify                          = false -> null
      - auto_qos_classify_police                   = false -> null
      - auto_qos_trust                             = false -> null
      - auto_qos_trust_cos                         = false -> null
      - auto_qos_trust_dscp                        = false -> null
      - auto_qos_video_cts                         = false -> null
      - auto_qos_video_ip_camera                   = false -> null
      - auto_qos_video_media_player                = false -> null
      - auto_qos_voip                              = false -> null
      - auto_qos_voip_cisco_phone                  = false -> null
      - auto_qos_voip_cisco_softphone              = false -> null
      - auto_qos_voip_trust                        = false -> null
      ~ description                                = "This is changed" -> "Test3"
        id                                         = "Cisco-IOS-XE-native:native/interface/GigabitEthernet=1%2F0%2F42"
      - ip_access_group_in_enable                  = false -> null
      - ip_access_group_out_enable                 = false -> null
      - ip_arp_inspection_trust                    = false -> null
      - ip_dhcp_snooping_trust                     = false -> null
      - ipv6_address_autoconfig_default            = false -> null
      - ipv6_address_dhcp                          = false -> null
      - ipv6_enable                                = false -> null
      - ipv6_nd_ra_suppress_all                    = false -> null
      - mab                                        = false -> null
      - mab_eap                                    = false -> null
        name                                       = "1/0/42"
      - shutdown                                   = false -> null
      + source_template                            = []
      - spanning_tree_portfast_trunk               = false -> null
      - speed_100                                  = false -> null
      - speed_1000                                 = false -> null
      - speed_10000                                = false -> null
      - speed_100000                               = false -> null
      - speed_2500                                 = false -> null
      - speed_25000                                = false -> null
      - speed_40000                                = false -> null
      - speed_5000                                 = false -> null
      - speed_nonegotiate                          = false -> null
        # (1 unchanged attribute hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.`

**Comment by danischm** [comment]
The import ID needs to be URL encoded, yes. For most of the boolean attributes the YANG model does not differentiate between a non-existing and `false` value, therefore we have to assume `false` when importing.

**Comment by NWA2000** [comment]
Gotcha, thank you for the info and fixing!

**Comment by danischm** [comment]
Right, we don't support multiple devices yet. During the import operation we do not have access to the config, meaning we would need to encode the device name in the "id" provided to "terraform import". Something like:

```
terraform import 'iosxe_interface_ethernet.interfaces["1/0/42"]' "device1;Cisco-IOS-XE-native:native/interface/GigabitEthernet=1%2F0%2F42"
```

I will re-open this issue to track this enhancement.
